### ISSUE TITLE
SearchIndex window is not defined

### ISSUE THREAD

**Issue by devinhalladay** [missive/emoji-mart#645]
Along the same lines as other issues with SSR and this library, I noticed that importing `SearchIndex` introduces a `window is not defined` error. I'm not quite sure how to deal with this, as I have a search function (headless search) that requires SearchIndex to be available on page load.

(Thanks for the cool library, I hope it's working better soon so I can put it to use :) )

**Comment by devinhalladay** [comment]
Here's the code I'm working with:

```tsx
import React, { useEffect, useRef } from 'react';
import data from '@emoji-mart/data';
import { SearchIndex } from 'emoji-mart';

function EmojiInput(props) {
  const ref = useRef(null);
  const results = useRef(null);
  const input = useRef<HTMLInputElement>(null);

  const [text, setText] = React.useState('Type something…');

  useEffect(() => {
    import('emoji-mart').then((EmojiMart) => {
      EmojiMart.init({ data });
      new EmojiMart.Picker({ ...props, data, ref });
    });
  }, []);

  async function search(value) {
    const searchResults = SearchIndex && (await SearchIndex.search(value));

    if (searchResults && results.current) {
      if (!searchResults.length) {
        setText('No results');
      } else {
        const val = searchResults.map((e) => e.skins[0].native).join(' ');
        setText(val);
      }
    }
  }

  return (
    <>
      <div ref={ref} />
      <div ref={results}>{text}</div>
      <input type="text" ref={input} onChange={(e) => search(e.target.value)} />
    </>
  );
}

export default EmojiInput;
```

I already tried dynamically importing the SearchIndex, but that doesn't work because if SearchIndex is loaded in a useEffect, my `search()` function won't be able to access it

**Comment by baptisteArno** [comment]
You can set your initialized `SearchIndex` in a state:

```ts
useEffect(() => {
    import('emoji-mart').then(({ init, SearchIndex }: any) => {
      init({ data })
      setSearchIndex(SearchIndex)
    })
  }, [])
```

Then use this state anywhere else to do the search 👍

**Comment by EtienneLem** [comment]
SSR issues have been addressed in **v5.2.0**. You can use [`@emoji-mart/react`](https://github.com/missive/emoji-mart#react) now.
